### ISSUE TITLE
Will not start cleaning

### ISSUE THREAD

**Issue by ahovis** [bwp91/homebridge-deebot#161]
### What issue do you have? Please be as thorough and explicit as possible.

The plugin installed with no problems. I am running a yeddi 2 pro vacuum and can access fine. Unfortunately, it will not start cleaning and does give an error. It will return to charge by the plugin if I start it using the yeedi app. I have attached the Homebridge log if that might help.

### Details of your setup.
* Do you use (1) Homebridge UI-X (2) Homebridge CLI or (3) HOOBS? 
Latest version of Homebridge 

* Which version of Homebridge/HOOBS do you have?

1.7.0. UI 4.55.1

* Which platform do you run Homebridge/HOOBS on (e.g. Raspberry Pi/Windows/HOOBS Box)? Please also mention your version of Node.js/NPM if known.
Raspberry Pi 4, Node.js is 18.19.0

* Which version of this plugin (homebridge-deebot) do you have? Has the issue started since upgrading from a previous version?
Version 6.1.1. Always had this problem.

* Which ECOVACS devices do you have that are causing issues? Please include product models if applicable.

[Deebot Error.pdf](https://github.com/bwp91/homebridge-deebot/files/14185315/Deebot.Error.pdf)

### Please paste any relevant logs below.

```

```

**Comment by MacDeal85** [comment]
I have the same problem. On Synology with Homebridge and "Yeedi Vac 2 Pro".

```html
{ type: 'vw', mid: '13666652' }
[09/03/2024, 20:58:36] [Deebot] [Robot] sent error [Cannot read properties of undefined
(reading 'forEach')].
event onCachedMapInfo
``` 

Is there anyone who has a solution for me. Thank you in advance. MacDeal

**Comment by mrbungle64** [comment]
@bwp91 

Can you please bump ecovacs-deebot.js Library to the latest beta (currently 0.9.6-beta.3)?
I'm not sure as I can't test it, but this issue should be fixed with this version.

**Comment by mrbungle64** [comment]
@bwp91 

Is 6.1.2-beta.0 ready to be tested?

**Comment by bwp91** [comment]
Hi @mrbungle64 yes this can be tried :)

**Comment by mrbungle64** [comment]
> Hi @mrbungle64 yes this can be tried :)

Great 👍🏻

@ahovis @MacDeal85 

Please try the beta and give feedback if the issue is solved

**Comment by MacDeal85** [comment]
Bei mir geht auch die Beta nicht. Laden ja, Cleaning nein.
<img width="876" alt="2024-03-17 11_33_01-" src="https://github.com/bwp91/homebridge-deebot/assets/162839362/223ad2f3-abda-492a-a547-ae5a292c1b0d">

**Comment by mrbungle64** [comment]
@MacDeal85 

> Bei mir geht auch die Beta nicht. Laden ja, Cleaning nein.

Der gemeldete Fehler (`[Deebot] [Robot] sent error [Cannot read properties of undefined
(reading 'forEach')].`) tritt aber nicht mehr auf, oder?

---

But the reported error  (`[Deebot] [Robot] sent error [Cannot read properties of undefined
(reading 'forEach')].`) no longer occurs, right?

**Comment by MacDeal85** [comment]
Nein, der scheint weg zu sein. Trotzdem startet er nicht bei Cleaning. zur Station fahren klappt einwandfrei.

**Comment by bwp91** [comment]
@mrbungle64 do you think this is an error thrown from your library? or mine? 

not following the german, please translate if anything important for me 😃

**Comment by mrbungle64** [comment]
@bwp91 

> @mrbungle64 do you think this is an error thrown from your library? or mine?
> 
> not following the german, please translate if anything important for me 😃

The originally reported error was thrown by my library.

> Nein, der scheint weg zu sein. Trotzdem startet er nicht bei Cleaning. zur Station fahren klappt einwandfrei.

Translation:
"No, the error no longer seems to occur. Nevertheless, it does not start with Cleaning. Returning to the (charging) station works perfectly."

---

I assume that this model also uses the V2 commands. I'll try to find information about it later.

**Comment by mrbungle64** [comment]
> I assume that this model also uses the V2 commands. I'll try to find information about it later.

I just updated the model dict information in the library and published 0.9.6-beta.4.

@ahovis @MacDeal85 
Please try if it's working after reinstalling the beta of homebridge-deebot. After reinstalling it should update to the latest beta of library.

**Comment by MacDeal85** [comment]
Perfect, works! Thank you!
